types/modules/channels.d.ts

```diff
@@ -0,0 +1,32 @@
+import type amqp = require('amqplib');
+
+interface ChannelConfig {
+  prefetch: number;
+}
+
+declare class Channels {
+  constructor(connection: amqp.Connection, config: ChannelConfig);
+  _connection: any;
+  _config: any;
+  _channels: Map<any, any>;
+  get(queue: any, config: any): Promise<any>;
+  defaultChannel(): Promise<any>;
+  /**
+   * Creates or returns an existing channel by it's key and config.
+   * @return {Promise} A promise that resolve with an amqp.node channel object
+   */
+  _get(key: any, config?: {}): Promise<any>;
+  _initNewChannel(key: any, config: any): Promise<any>;
+}
+
+declare class ChannelAlreadyExistsError extends Error {
+  constructor(name: any, config: any);
+  name: any;
+  config: any;
+}
```

Comment:
```suggestion
  constructor(name: string, config: any);
  public readonly name: string;
  public readonly config: any;
}
```

and please type config

Reply:
@ramhr what about this ?